### PR TITLE
Feat/hierarchy flat

### LINKED ISSUE #5: Add hierarchy flat
Add `heirarchyFlat` field to each user object in `getById` and `searchByName` functions. 
The `heirarchyFlat` field will be a concatenation of the hierarchy field's string array and the job field.

### PR DESCRIPTION
**Pull request by lustigy96**
I added the flat hierarchy feature using a 'normalization' function, which normalize the data from Kartoffel to the required data in friends.

### CODE REVIEW

File: src/utils/normalize.data.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { IUser, IUserNormalized } from '../user/user.interface';
+
```
Comment (yonatandt) [inline src/utils/normalize.data.ts:2]: Fix all `normalised` to `normalized`.

File: src/utils/normalize.data.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+export const getNormalisedUser = (user : IUser) :IUserNormalized => {
```
Comment (yonatandt) [inline src/utils/normalize.data.ts]: If that's the only export use `export default`

Comment (yonatandt) [reply src/utils/normalize.data.ts]: Add documentation

Comment (yonatandt) [reply src/utils/normalize.data.ts]:
```suggestion
export const getNormalisedUser = (user: IUser): IUserNormalized => {
```

File: src/user/kartoffel.service.ts [hunk 3]
```diff
@@ -55,7 +56,9 @@ export default abstract class KartoffelService {
     } catch (err) {
       throw new ApplicationError(`Unknown Error: ${err} `);
     }
-    const users: IUser[] = res.data;
+    const users: IUserNormalized[] = res.data.map((user: IUser) => {
+      return getNormalisedUser(user);
+    });
```
Comment (yonatandt) [inline src/user/kartoffel.service.ts]:
```suggestion
    const users: IUserNormalized[] = res.data.map((user: IUser) => getNormalisedUser(user));
```

**Review by yonatandt — CHANGES_REQUESTED** [review]

File: src/utils/normalize.data.ts (hunk 1 above)
Comment (lustigy96) [reply src/utils/normalize.data.ts:2]: fixed ;)

File: src/utils/normalize.data.ts (hunk 2 above)
Comment (lustigy96) [reply src/utils/normalize.data.ts]: accepted

**Review by yonatandt — APPROVED** [review]

**Pull request merged**